pennylane/tape/tape.py

```diff
@@ -703,9 +703,9 @@ def inv(self):
         self._par_info = {parameter_mapping[k]: v for k, v in self._par_info.items()}
 
         for idx, op in enumerate(self._ops):
-            try:
+            if op.has_adjoint:
                 self._ops[idx] = op.adjoint()
-            except qml.operation.AdjointUndefinedError:
```

Comment:
This is probably out of date.  We want to avoid using in-place inversion in PennyLane, as it is now deprecated and will be removed in 0.28.